apps/gateway/test/champions_test.exs

```diff
@@ -0,0 +1,286 @@
+defmodule Gateway.Test.Champions do
+  @moduledoc """
+  Test for Champions of Mirra messages.
+  """
+  use ExUnit.Case
+
+  alias GameBackend.Users.Currencies.CurrencyCost
+  alias Champions.{Units, Users, Utils}
+  alias GameBackend.Users.Currencies
+
+  alias Gateway.Serialization.{
+    Currency,
+    Error,
+    Unit,
+    UnitAndCurrencies,
+    User,
+    UserCurrency,
+    WebSocketResponse
+  }
+
+  alias Gateway.SocketTester
+
+  # import Plug.Conn
+  # import Phoenix.ConnTest
+
+  setup_all do
+    # Start Phoenix endpoint
+    {:ok, _} =
+      Plug.Cowboy.http(Gateway.Endpoint, [],
+        ip: {127, 0, 0, 1},
+        port: 4001,
+        dispatch: [
+          _: [{"/2", Gateway.ChampionsSocketHandler, []}]
+        ]
+      )
+
+    :ok
+  end
+
+  setup do
+    {:ok, socket_tester} = SocketTester.start_link()
+
+    {:ok, %{socket_tester: socket_tester}}
+  end
+
+  describe "users" do
+    test "users", %{socket_tester: socket_tester} do
+      # Create our user
+      :ok = SocketTester.create_user(socket_tester, "Username")
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{}}}
+
+      fetch_last_message(socket_tester)
+      %WebSocketResponse{response_type: {:user, user}} = get_last_message()
+
+      # Creating another user with the same name fails
+      :ok = SocketTester.create_user(socket_tester, "Username")
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:error, %Error{reason: "username_taken"}}}
+
+      # Get user by name
+      :ok = SocketTester.get_user_by_username(socket_tester, "Username")
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, ^user}}
```

Comment:
This would fail if the `Users.get_user_by_username\1` method change their preloads right?

Reply:
Do you mean if we preloaded different things in get_user and get_user_by_username? I think that should not happen in the first place

Reply:
Oh sorry, i didn't noticed that the register function uses `get_user` to retrieve the user after you insert it